src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -1,926 +1,159 @@
-      doc.defaultView.openTrustedLinkIn("https://monitor.firefox.com/scan", "tab", { postData });
-    } else {
-      doc.defaultView.openTrustedLinkIn("https://monitor.firefox.com/", "tab", {});
-    }
-    FirefoxMonitor.notifyEventListeners(`${gNotificationID}_submit${checkboxChecked ? "_checked" : ""}`);
-  }
-
-  clearInvalidMessage(textbox);
-  // Make sure we don't show the "x" button, it's problematic because it fires
-  // a command event that we can't really distinguish from an "enter" keypress.
-  textbox._searchIcons.selectedIndex = 0;
-  const button = doc.getAnonymousElementByAttribute(
-    doc.getElementById(`${gNotificationID}-notification`), "anonid", "button");
-  const evtWasCommand = event.type === "command";
-  const email = textbox.value;
-  if (email && isEmailValid(email)) {
-    if (evtWasCommand) {
-      submit(email);
-      return;
-    }
-    button.removeAttribute("disabled");
-    return;
-  }
-  if (evtWasCommand) {
-    showInvalidMessage(textbox);
-  }
-  button.setAttribute("disabled", "true");
-};
-
-
 this.FirefoxMonitor = {
-  init(aExtension, aVariation, warnedSites, firstRunTimestamp) {
-    gExtension = aExtension;
-    gFirstRunTimestamp = firstRunTimestamp;
-
-    try {
-      const pref = Preferences.get(`extensions.fxmonitor_shield_mozilla_org.variationName`, null);
-      UI_VARIANT = parseInt(pref !== null ? pref : aVariation);
-    } catch (e) {
-      Cu.reportError("Invalid variation specified, exiting.");
-      return;
-    }
+  domainMap: new Map(),
+  warnedHostSet: new Set(),
+  extension: null,
+  observerAdded: false,
+  kNewtabURL: "about:newtab",
+  kDisabledPref: "browser.firefoxMonitor.disabled",
+
+  init(aExtension, warnedSites) {
+    this.extension = aExtension;
+    Services.scriptloader.loadSubScript(
+      this.getURL("privileged/blurts/subscripts/Globals.jsm"));
+    Services.scriptloader.loadSubScript(
+      this.getURL("privileged/blurts/subscripts/EveryWindow.jsm"));
+    Services.scriptloader.loadSubScript(
+      this.getURL("privileged/blurts/subscripts/PanelUI.jsm"));
 
     if (warnedSites) {
-      warnedHostSet = new Set(warnedSites);
+      this.warnedHostSet = new Set(warnedSites);
     }
 
-    fetch(gExtension.getURL("breaches.json")).then(function(response) {
+    fetch(this.getURL("breaches.json")).then((response) => {
       return response.json();
-    }).then(function(sites) {
+    }).then((sites) => {
       for (let site of sites) {
-        domainMap.set(site.Domain.toLowerCase(), { Domain: site.Domain, Name: site.Name, Title: site.Title, PwnCount: site.PwnCount, BreachDate: site.BreachDate, DataClasses: site.DataClasses, logoSrc: `${site.Name}.${site.LogoType}` });
+        this.domainMap.set(site.Domain.toLowerCase(), { Domain: site.Domain, Name: site.Name, Title: site.Title, PwnCount: site.PwnCount, BreachDate: site.BreachDate, DataClasses: site.DataClasses.join(", "), logoSrc: `${site.Name}.${site.LogoType}` });
```

Comment:
super nit: Not sure if we want this to be "LogoSrc" to be consistent with the other "Domain" and "PwnCount" variable style.